src/simulators/unitary/unitarymatrix.hpp

```diff
@@ -0,0 +1,232 @@
+/**
+ * Copyright 2018, IBM.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 found in
+ * the LICENSE.txt file in the root directory of this source tree.
+ */
+
+#ifndef _qv_unitary_matrix_hpp_
+#define _qv_unitary_matrix_hpp_
+
+
+#include "framework/utils.hpp"
+#include "simulators/statevector/qubitvector.hpp"
+
+namespace QV {
+
+//============================================================================
+// UnitaryMatrix class
+//============================================================================
+
+// This class is derived from the QubitVector class and stores an N-qubit 
+// matrix as a 2*N-qubit vector.
+// The vector is formed using column-stacking vectorization as under this
+// convention left-matrix multiplication on qubit-n is equal to multiplication
+// of the vectorized 2*N qubit vector also on qubit-n.
+
+template <class data_t = complex_t*>
+class UnitaryMatrix : public QubitVector<data_t> {
+
+public:
+  // Type aliases
+  using BaseVector = QubitVector<data_t>;
+
+  //-----------------------------------------------------------------------
+  // Constructors and Destructor
+  //-----------------------------------------------------------------------
+
+  UnitaryMatrix() : UnitaryMatrix(0) {};
+  explicit UnitaryMatrix(size_t num_qubits);
+  UnitaryMatrix(const UnitaryMatrix& obj) = delete;
```

Comment:
Just asking: Why do we explicitly remove copy constructor? I guess because we don't want copy from other UnitaryMatrixes, but is there a reason?